scripts/us_census/acs5yr/subject_tables/common/generate_col_map.py

```diff
@@ -0,0 +1,141 @@
+"""Module to generate a column map to their corresponding statistical variable defintions."""
+
+import os
+import json
+
+import pandas as pd
+
+#TODO:Remove this import and make it from ~/data/utils/ directory when it is available
+os.system(
+    'curl https://raw.githubusercontent.com/Abilityguy/data/dcid_gen/util/statvar_dcid_generator.py -LO statvar_dcid_generator.py'
+)
+from statvar_dcid_generator import get_stat_var_dcid
+
+
+
+def generate(specDict: dict, columnList: list) -> dict:
+    """Wrapper function for generateColMapBase class to generate column map.
+
+  Args:
+    specDict:
+      A dictionary containing specifications for the different properties of the statistical variable.
+    columnList:
+      A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+
+  Returns:
+    A dict mapping each column to their respective stat_var node definitions.
+  """
+    col_map_obj = GenerateColMapBase(specDict=specDict, columnList=columnList)
+    return col_map_obj._generate_stat_vars_from_spec()
+
+
+class GenerateColMapBase:
+    """module to generate a column map given a list of columns of the dataset and a JSON Spec
+
+  Attributes:
+    specDict: A dictionary containing specifications for the different properties of the statistical variable.
+    columnList: A list of column names for which the column map needs to be generated. This is typically the column header in the dataset.
+  """
+
+    def __init__(self, specDict=None, columnList=None):
```

Comment:
what is the impact of having default None arguments for spec and column?

Reply:
use snake case for function_parameter_name

Reply:
No impact with using None. Perhaps, it would be better to default the arguments as `spec_dict={}, column_list=[]` to make it more intuitive for someone reading the code. Shall we do that?